src/compiler/factory/emitHelpers.ts

```diff
@@ -657,6 +886,28 @@ namespace ts {
             };`
     };
 
+    export const propKeyHelper: UnscopedEmitHelper = {
+        name: "typescript:propKey",
+        importName: "__propKey",
+        scoped: false,
+        text: `
+            var __propKey = (this && this.__propKey) || function (x) {
+                return typeof x === "symbol" ? x : "".concat(x);
+            };`

```

Comment:
The helper will throw if `x` has a `toPrimitive` method returning a symbol, e.g.
```js
x = { [Symbol.toPrimitive]: () => Symbol.toPrimitive }
```

Reply:
We issue diagnostics if the property name is not `string`, `number`, or `symbol`, so I'm not sure we need to go to that level of effort in the helper.